support/cas-server-support-redis-core/src/main/java/org/apereo/cas/redis/core/util/RedisUtils.java

```diff
@@ -0,0 +1,55 @@
+package org.apereo.cas.redis.core.util;
+
+import lombok.experimental.UtilityClass;
+import lombok.val;
+import org.apache.commons.io.IOUtils;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.data.redis.core.ScanOptions;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * This is {@link RedisUtils}.
+ *
+ * @author Fireborn Z
+ * @since 6.5.0
+ */
+@UtilityClass
+public class RedisUtils {
+    private static final long SCAN_COUNT = 1000L;
+
+    /**
+     * Get redis keys using scan command.
+     *
+     * @param redisTemplate the redisTemplate
+     * @param pattern       the redis keys pattern
+     * @return the redis keys
+     */
+    public static Set<String> keys(final RedisTemplate<String, ?> redisTemplate, final String pattern) {
```

Comment:
Indicate the operation only uses a count of 1000 in the javadocs and reference the constant.

Reply:
I also dont think this method is necessary; should be removed.

Reply:
A proper count value can greatly reduce the number of interactions with the Redis, the premise is to have a correct estimate of the amount of data in redis. I prefer to use a slightly larger default value to ensure that the program will not have obvious performance problems in large data scenarios, but the caller can also specify other values.

Reply:
Regardless of what value you pick, that number would never be a "correct estimate", and, there are no "callers" in CAS. We are not designing an SDK; this is a platform and a product. Assumptions like this need to be made at a level where a call is made by the user in a way that is usable and configurable. A method that takes this assumption should not be available by default, and whoever wants to make that call at a higher level can take the resulting stream and narrow and filter it as necessary.

Reply:
How about this, put scanCount parameter as a redis property in BaseRedisProperties, so that users can configure it for different modules in different scenarios to achieve performance optimization purposes. In that way, RedisUtils shoud be a bean.

Reply:
This method has been removed, pls review.